app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -101,16 +101,38 @@ def csv_file_download_path(run)
       )
     end
 
-    # Return the appropriate field tag for the parameter, based on its type.
-    # If the parameter has a `validates_inclusion_of` validator, return a dropdown list of options instead.
-    def parameter_field(form_builder, parameter_name)
-      inclusion_validator = form_builder.object.class.validators_on(parameter_name).find do |validator|
+    # Returns the appropriate value for validator's inclusion options for a Proc,
+    # lambda, symbol, or other types.
+    #
+    # If there is a value returned, the dropdown list of options will be used.
+    #
+    # Procs, lambdas, and symbols options are not supported by the dropdown form,
+    # and return nil.
+    #
+    # @param object [Task] The object for which the value needs to be resolved.
+    # @param parameter_name [String] The parameter name.
+    #
+    # @return [Object] value of the resolved inclusion option.
+    def resolve_inclusion_value(object, parameter_name)
+      inclusion_validator = object.class.validators_on(parameter_name).find do |validator|
         validator.kind == :inclusion
       end
+      return unless inclusion_validator
 
+      in_option = inclusion_validator.options[:in] || inclusion_validator.options[:within]
+
+      return if in_option.respond_to?(:call) || in_option.respond_to?(:to_sym)
```

Comment:
I was thinking that we should actually test for the case we want instead of the cases we don't want. But it's probably fine that way.

Reply:
Ah sorry - Misunderstood, thanks! And good catch on the Range edge case, handling for what's supported is definitely easier to manage than filtering out what isn't supported.